components/EditDatesForm.js

```diff
@@ -0,0 +1,85 @@
+import styled from "styled-components";
+import Image from "next/image";
+import saveSvg from "../assets/save.svg";
+import { useState } from "react";
+
+const timestampToIsoString = (timestamp) => new Date(timestamp * 1000).toISOString().substring(0, 10);
+
+export default function EditDatesForm({ destination, onUpdateDetail, toggleModal }) {
+  const [startDate, setStartDate] = useState(timestampToIsoString(destination.startDate));
+  const [endDate, setEndDate] = useState(timestampToIsoString(destination.endDate));
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onUpdateDetail({
+      startDate: Date.parse(startDate) / 1000,
+      endDate: Date.parse(endDate) / 1000,
+    });
+    toggleModal();
```

Comment:
One more thing: It is unnecessary to handle the toggleModal prop to this component adn trust that it does use the function at the same time it uses the onUpdateDetail function. A design like this would be better:
```
export default function EditDatesForm({ destination, onUpdateDetail }) {
  const [startDate, setStartDate] = useState(timestampToIsoString(destination.startDate));
  const [endDate, setEndDate] = useState(timestampToIsoString(destination.endDate));

  const handleSubmit = (event) => {
    event.preventDefault();
    onUpdateDetail(Date.parse(startDate) / 1000, Date.parse(endDate) / 1000);
}
```

And you could call the component like this:
```
<EditDatesForm
              destination={destination}
              onUpdateDetail={(startDate, endDate) => {toggleModal(); onUpdateDetail({startDate, endDate})}}
            />
```